src/CachePlugin.php

```diff
@@ -67,12 +68,16 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
 
         if ($cacheItem->isHit()) {
             // return cached response
-            return new FulfilledPromise($cacheItem->get());
+            $data = $cacheItem->get();
+            $response = $data['response'];
+            $response = $response->withBody($this->streamFactory->createStream($data['body']));
```

Comment:
You are right. Initially, it was the plan to return null when the body is empty. All these PSR7 versions in my head sometimes make me wonder about the REAL interfaces.